src/org/sosy_lab/java_smt/solvers/cvc5/CVC5InterpolatingProver.java

```diff
@@ -0,0 +1,370 @@
+
+public class CVC5InterpolatingProver extends CVC5AbstractProver<Term>
+    implements InterpolatingProverEnvironment<Term> {
+
+  private final FormulaManager mgr;
+  private final Set<ProverOptions> solverOptions;
+  private final int seed;
+
+  CVC5InterpolatingProver(
+      CVC5FormulaCreator pFormulaCreator,
+      ShutdownNotifier pShutdownNotifier,
+      @SuppressWarnings("unused") int randomSeed,
+      Set<ProverOptions> pOptions,
+      FormulaManager pMgr) {
+    super(pFormulaCreator, pShutdownNotifier, randomSeed, pOptions, pMgr);
+    mgr = pMgr;
+    solverOptions = pOptions;
+    seed = randomSeed;
+  }
+
+  /**
+   * Sets the same solver Options of the Original Solver to the separate solvertoSet, except for
+   * produce-interpolants which is set here. From CVC5AbstractProver Line 66
+   */
+  private void setSolverOptions(int randomSeed, Set<ProverOptions> pOptions, Solver solvertoSet) {
+    if (incremental) {
+      solvertoSet.setOption("incremental", "true");
+    }
+    if (pOptions.contains(ProverOptions.GENERATE_MODELS)) {
+      solvertoSet.setOption("produce-models", "true");
+    }
+    if (pOptions.contains(ProverOptions.GENERATE_UNSAT_CORE)) {
+      solvertoSet.setOption("produce-unsat-cores", "true");
+    }
+    solvertoSet.setOption("produce-assertions", "true");
+    solvertoSet.setOption("dump-models", "true");
+    solvertoSet.setOption("output-language", "smt2");
+    solvertoSet.setOption("seed", String.valueOf(randomSeed));
+    solvertoSet.setOption("produce-interpolants", "true");
+
+    // Set Strings option to enable all String features (such as lessOrEquals)
+    solvertoSet.setOption("strings-exp", "true");
+
+    // Enable more complete quantifier solving (for more info see CVC5QuantifiedFormulaManager)
+    solvertoSet.setOption("full-saturate-quant", "true");
+  }
+
+  @Override
+  public Term addConstraint(BooleanFormula pConstraint) throws InterruptedException {
+    Preconditions.checkState(!closed);
+    Term t = creator.extractInfo(pConstraint);
+
+    super.addConstraint(pConstraint);
+    return t; // t is not returned in the Abstract Class
+  }
+
+  @Override
+  public BooleanFormula getInterpolant(Collection<Term> pFormulasOfA)
+      throws SolverException, InterruptedException {
+    Preconditions.checkState(!closed);
+
+    if (pFormulasOfA.isEmpty()) { // Catch trivial case
+      return mgr.getBooleanFormulaManager().makeBoolean(true);
+    }
+
+    // formulasOfB := assertedFormulas - pFormulasOfA
+    Set<Term> formulasOfB =
+        assertedFormulas.stream()
+            .flatMap(c -> c.stream())
+            .filter(n -> !pFormulasOfA.contains(n))
+            .collect(ImmutableSet.toImmutableSet());
+
+    if (formulasOfB.isEmpty()) { // Catch trivial case
+      return mgr.getBooleanFormulaManager().makeBoolean(false);
+    }
+
+    // fit the Input to work with getCVC5Interpolation
+
+    ImmutableList<Collection<Term>> formAAsList = ImmutableList.of(pFormulasOfA);
```

Comment:
Is there a reason for having "lists" instead of "sets"?

Reply:
No :) its replaced.